setup.py

```diff
@@ -23,12 +17,14 @@
         metaflow=metaflow.main_cli:main
       ''',
       install_requires = [
-        'click>=7.0',
+        'click>=7.0,<8',
         'requests',
         'boto3',
-        'pylint<2.5.0',
-        'kfp'
+        'pylint<2.5.0'
       ],
       tests_require = [
         'coverage'
-      ])
+      ],
+      extras_require = {
+        'kfp': 'zillow-kfp'
```

Comment:
Why is `kfp` and optional library again? The Metaflow kfp plugin explicitly imports the kfp sdk.

Reply:
But the `kfp` plugin itself is an optional part of `metaflow`, thus why this is an extra. This was mainly how I fixed the upstream `metaflow` tests that run via Github Actions; they were failing as they couldn't find `zillow-kfp` (as we don't publish it publicly). So this was the best option to make it stop failing.

If we want to include this as a regular dependency thats fine, but obviously we'll need to rethink our publishing strategy for `zillow-kfp`.